### ISSUE TITLE
Browser hangs with 100% CPU load

### ISSUE THREAD

**Issue by ctron** [mattermost/platform#3545]
For feature requests, please see: http://www.mattermost.org/feature-requests/.  
For troubleshooting questions, please post on the forums: http://forum.mattermost.org/.

For issues with the product, please follow the steps at http://www.mattermost.org/filing-issues/ and post here in the following format: 

#### Summary
Switching to a specific channel causes browsers to hang on 100% CPU load

#### Steps to reproduce
Join the channel: https://mattermost.eclipse.org/eclipse/channels/platform
Mattermost: 3.1.0

#### Expected behavior
Not to cause 100% load, not to crash browser

#### Observed behavior
...

#### Possible fixes
...

**Comment by ctron** [comment]
Restarting the mattermost server did not help.

**Comment by hmhealey** [comment]
Hi @ctron,

There's an issue in Mattermost 3.1 that causes certain invalid URLs to cause the Markdown parser to enter an infinite loop. You need a system admin to delete the offending post using the steps outlined [here](https://github.com/mattermost/platform/issues/3407#issuecomment-228636464). Also, if you wouldn't mind posting the bad link here so that I can confirm it's fixed in our next version?

**Comment by ctron** [comment]
Unfortunately this fix seems not to work. Once I enabled the Developer Mode, I don't see an option to disable formatting.

**Comment by hmhealey** [comment]
Oh sorry about that. I edited that post a few times to reflect some differences to those steps for our 3.2 version which made the first step incorrect. You actually need to enable Testing to get access to the ability to disable formatting. I've updated it so that it makes sense again.

**Comment by ctron** [comment]
I just checked the channel, can you give me a hint on what to delete? I got quite a few links in there...

**Comment by ctron** [comment]
The issue was actually the channel header:

```
[![Eclipse Platform](https://eclipse.org/eclipse.org-common/themes/solstice/public/images/favicon.ico)](http://eclipse.org/eclipse) | [![Gerrit](https://git.eclipse.org/r/favicon.ico)](https://git.eclipse.org/r/#/q/status:open+label:Verified%252B1+-label:Code-Review%253C%253D-1+(projects:platform+OR+projects:e4)
|[:closed_book:](https://wiki.eclipse.org/Platform)|
[![Build Status](https://img.shields.io/jenkins/s/https/hudson.eclipse.org/shared/ep46N-unit-win32.svg?label=win)](https://hudson.eclipse.org/shared/view/Eclipse%20and%20Equinox/job/ep46N-unit-win32/)
[![Build Status](https://img.shields.io/jenkins/s/https/hudson.eclipse.org/shared/ep46N-unit-lin64.svg?label=linux)](https://hudson.eclipse.org/shared/view/Eclipse%20and%20Equinox/job/ep46N-unit-lin64/)
[![Build Status](https://img.shields.io/jenkins/s/https/hudson.eclipse.org/shared/ep46N-unit-mac64.svg?label=osx)](https://hudson.eclipse.org/shared/view/Eclipse%20and%20Equinox/job/ep46N-unit-mac64/)
```

**Comment by hmhealey** [comment]
Awesome, thanks for the report. I've confirmed that the issue is fixed on 3.2 so it shouldn't happen again.
